src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -1086,6 +1088,23 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
     }
   }
 
+  public <E> void assertNoneSatisfy(AssertionInfo info, Iterable<? extends E> actual, Consumer<? super E> requirements) {
+    assertNotNull(info, actual);
+    requireNonNull(requirements, "The Consumer<T> expressing the assertions requirements must not be null");
+    boolean anyMatch = stream(actual.spliterator(), false).anyMatch(e -> {
+      try {
+        requirements.accept(e);
+      } catch (AssertionError ex) {
+        return false;
```

Comment:
instead of returning a boolean, let's throw a `noElementsShouldSatisfy` failure with the element that should not have met the given requirements.

Reply:
i refactored it once again, and even found a bug and adapted the test...